tests/api/controllers/connector/connector.c

```diff
@@ -22,9 +22,19 @@ int main(int argc, char **argv) {
       ts_assert_int_equal(wb_connector_get_presence(rear_connector), 1, "connector presence should be 1");
       wb_connector_unlock(front_connector);
       wb_connector_unlock(rear_connector);
-    } else if (t == 38 * TIME_STEP) {
+    }  
+    if (t == 35 * TIME_STEP) {
+      ts_assert_boolean_equal(wb_connector_is_locked(rear_connector), true, "connector presence should be locked");
+
+    }  
+    if (t == 38 * TIME_STEP) {
       ts_assert_int_equal(wb_connector_get_presence(rear_connector), 0,
                           "connector presence should be 0, as the other module should have fallen.");
+      wb_connector_unlock(rear_connector);
+
+    } 
+    if (t == 39 * TIME_STEP) {
+      ts_assert_boolean_equal(wb_connector_is_locked(rear_connector), true, "connector presence should be locked");
```

Comment:
```suggestion
    } else if (t == 35 * TIME_STEP)
      ts_assert_boolean_equal(wb_connector_is_locked(rear_connector), true, "connector presence should be locked");
    else if (t == 38 * TIME_STEP) {
      ts_assert_int_equal(wb_connector_get_presence(rear_connector), 0,
                          "connector presence should be 0, as the other module should have fallen.");
      wb_connector_unlock(rear_connector);
    } else if (t == 39 * TIME_STEP) {
      ts_assert_boolean_equal(wb_connector_is_locked(rear_connector), true, "connector presence should be locked");
```

The test if failing because there are some trailing whitespaces.